src/include/migraphx/shape.hpp

```diff
@@ -156,8 +156,33 @@ struct shape
 
     shape(const std::vector<shape>& subs);
 
+    /**
+     * Creates an output shape with dimensions equal to the input lengths and strides determined
+     * by the permutation argument.
```

Comment:
Strides are created so that `find_permutation` will return the permutation that is passed.